usr.bin/paste/paste.c

```diff
@@ -98,51 +114,50 @@ main(int argc, char *argv[])
 		delim = tab;
 	}
 
-	if (seq)
-		rval = sequential(argv);
-	else
-		rval = parallel(argv);
+	for (filecnt = 0; *argv; ++argv, ++filecnt) {
```

Comment:
Have you considered using fileargs?
man.freebsd.org/cgi/man.cgi?query=cap_fileargs&sektion=3&manpath=freebsd+14.1-release+and+ports
Opening files up front has some advantages, it's faster.
I wonder why you decided to open files upfront and not use fileargs?

Reply:
Lovely! I didn't know that, Mariusz. Thanks a lot, I'll take care of it.